docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,485 @@
+# EP: Definition of the Fleet and Device API spec
+<!-- this format is inspired by the K8S KEP format https://raw.githubusercontent.com/kubernetes/enhancements/master/keps/NNNN-kep-template/README.md -->
+## Summary
+
+The Device API, and Fleet API, and the dynamics of managing individual
+devices as well as fleets of devices is a fundamental piece of flightctl.
+
+The Fleet API must serve as a way to manage groups of devices while still allowing
+for the management of individual devices in a way that we can integrate with other
+management systems.
+
+## Motivation
+
+We need agreement on the API design direction and the semantics of the API to be able to
+implement and grow the flightctl service.
+
+### Goals
+
+* Define the format and behavior of the Device and Fleet APIs
+
+* Allow the management of individual fleetless devices on a smaller scale,
+  as an integration mechanism for services like AAP and ACM.
+
+* Allow the management of fleets of devices on a larger scale, via gitops or
+  API.
+
+* Change API fields to provide a more consistent experience.
+
+### Non-Goals
+
+
+
+## Proposal
+
+### Device API
+
+The device API is divided into multiple methods, some user facing, and other device/facing,
+this design facilitates cleaner RBAC control of the endpoint access. The diagram is only
+intended to illustrate the flow of information between the user/system and the device.
+
+```mermaid
+block-beta
+columns 1
+  user(["User"])
+
+  space
+
+
+  block
+    frontEndpoint["/device/{name}"]
+  end
+
+  user --> frontEndpoint
+  frontEndpoint --> user
+
+  space
+  block
+    metadata
+    devSpec["spec"]
+    devStatus["status"]
+  end
+
+  space
+
+  frontEndpoint --> metadata
+  frontEndpoint --> devSpec
+
+  block
+     renderedEndpoint["/device/{name}/rendered"]
+     statusEndpoint["/device/{name}/status"]
+  end
+
+  space
+
+  statusEndpoint --> devStatus
+  statusEndpoint --> devStatus
+  devSpec --> renderedEndpoint
+  renderedEndpoint --> Device
+  Device --> statusEndpoint
+
+
+  Device(["Device"])
+
+  style frontEndpoint fill:#969,stroke:#333,stroke-width:4px
+  style renderedEndpoint fill:#969,stroke:#333,stroke-width:4px
+  style statusEndpoint fill:#969,stroke:#333,stroke-width:4px
+  ```
+
+#### `/device/{name}` User facing method
+
+This endpoint can be accessed by the user or external systems to
+manage individual devices, also internally from the fleet specifications
+for devices.
+
+The proposed changes include:
+
+* Reincorporating device.spec as a visible part through this frontend API.
+
+* Adding a managementMode field to the device.spec, this field can be set to "fleet" or "manual".
```

Comment:
```suggestion
```

After RFE spec, we wouldn't need this field.

Everything would be controlled by the metadata.owner field, and the "flightctl.io/fleet-controller=paused" label